iohelper/iohelper.go

```diff
@@ -0,0 +1,95 @@
+package iohelper
+
+/*
+ * This file contains structs and functions related to interacting with files
+ * and directories.
+ */
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/greenplum-db/gp-common-go-libs/gplog"
+	"github.com/greenplum-db/gp-common-go-libs/operating"
+	"github.com/pkg/errors"
+)
+
+/*
+ * Generic file/directory manipulation functions that use the function pointers
+ * and types from the operating package, for easier test mocking around file IO.
+ */
+
+func OpenFileForReading(filename string) (operating.ReadCloserAt, error) {
```

Comment:
I'm not sure these functions OpenFileForReading and OpenFileForWriting provide much value over the direct OpenFileRead and OpenFileWrite functions we already have. I would vote not to add these.